video/decode/vd_lavc.c

```diff
@@ -485,6 +489,15 @@ static void select_and_set_hwdec(struct mp_filter *vd)
 
         if (!hwdec_requested) {
             MP_VERBOSE(vd, "No hardware decoding requested.\n");
+            /*
+             * Reset attempted hwdecs so that if the hwdec list is reconfigured
+             * we attempts all of them from the beginning. The most practical
+             * reason for this is that Ctrl+h toggles between `no` and `auto`,
+             * and we want `auto` to reevaluate from a clean slate each time.
```

Comment:
These probably should be updated to `auto-safe` as well.

Reply:
Fixed.